src/go/rdctl/cmd/start.go

```diff
@@ -0,0 +1,178 @@
+			return fmt.Errorf("start command: unrecognized command-line arguments specified: %v", args)
+		}
+		return doStartOrSetCommand(cmd)
+	},
+}
+
+var applicationPath string
+
+func init() {
+	rootCmd.AddCommand(startCmd)
+	updateCommonStartAndSetCommands(startCmd)
+	startCmd.Flags().StringVarP(&applicationPath, "path", "p", "", "Path to main executable.")
+}
+
+// Unavoidable race condition here.
+// There's no system-wide mutex that will let us guarantee that if rancher desktop is running when
+// we test it (easiest to just try to get the settings), that it will still be running when we
+// try to upload the settings (if any were specified).
+func doStartOrSetCommand(cmd *cobra.Command) error {
+	_, err := getListSettings()
+	if err == nil {
+		if applicationPath != "" {
+			// `--path | -p` is not a valid option for `rdctl set...`
+			return fmt.Errorf("--path specified but Rancher Desktop is already running", applicationPath)
+		}
+		return doSetCommand(cmd)
+	}
+	return doStartCommand(cmd)
+}
+
+func doStartCommand(cmd *cobra.Command) error {
+	var commandLineArgs []string
+
+	if cmd.Flags().Changed("container-engine") {
+		commandLineArgs = append(commandLineArgs, "--kubernetes-container-engine", specifiedSettings.ContainerEngine)
+	}
+	if cmd.Flags().Changed("kubernetes-enabled") {
+		commandLineArgs = append(commandLineArgs, "--kubernetes-enabled", strconv.FormatBool(specifiedSettings.Enabled))
+	}
+	if cmd.Flags().Changed("kubernetes-version") {
+		commandLineArgs = append(commandLineArgs, "--kubernetes-version", specifiedSettings.Version)
+	}
+	if cmd.Flags().Changed("flannel-enabled") {
+		commandLineArgs = append(commandLineArgs, "--kubernetes-options-flannel", strconv.FormatBool(specifiedSettings.Flannel))
+	}
+	if applicationPath == "" {
+		pathLookupFuncs := map[string]func() string{
+			"windows": getWindowsRDPath,
+			"linux":   getLinuxRDPath,
+			"darwin":  getMacOSRDPath,
+		}
+		getPathFunc, ok := pathLookupFuncs[runtime.GOOS]
+		if !ok {
+			return fmt.Errorf("don't know how to find the path to Rancher Desktop on OS %s", runtime.GOOS)
+		}
+		applicationPath = getPathFunc()
+		if applicationPath == "" {
+			return fmt.Errorf("no executable found in the default location; please retry with the --path|-p option")
+		}
+	}
+	return launchApp(applicationPath, commandLineArgs)
+}
+
+func launchApp(applicationPath string, commandLineArgs []string) error {
+	var commandName string
+	var args []string
+
+	if runtime.GOOS == "darwin" {
+		commandName = "open"
+		args = append(args, "-a", applicationPath)
+		if len(commandLineArgs) > 0 {
+			args = append(args, "--args")
+			args = append(args, commandLineArgs...)
+		}
+	} else {
+		commandName = applicationPath
+		args = commandLineArgs
+	}
+	// Include this output because there's a delay before the UI comes up.
+	// Without this line, it might look like the command doesn't work.
+	fmt.Fprintf(os.Stderr, "About to launch %s %s ...\n", commandName, strings.Join(args, " "))
```

Comment:
I prefer my way. The `About to launch` makes more sense when you finally see the UI appear. And why not show the arguments? Because that shows the user how they can launch the app directly from the command-line without relying on `rdctl start`, like a view-source for macOS users.